### PR TITLE
feat: chunk branchCommits to prevent timeouts

### PR DESCRIPTION
**Pull request by codebytere**
This PR chunks commits to mitigate the GitHub API weirdness that occurs past a certain threshold of commits. 

/cc @MylesBorins @BethGriggs

### CODE REVIEW

File: branch-diff.js [hunk 1]
```diff
@@ -87,6 +87,7 @@ function diffCollected (options, branchCommits, callback) {
   }
 
   let list = branchCommits[1].filter((commit) => !isInList(commit))
+  list = list.slice(2000, 2500)
```
Comment (MylesBorins) [inline branch-diff.js]: so this won't work as a permanent fix. This needs to be hand spun to change the range for each set of commits we are going to audit

0, 500
500, 1000
1000, 1500

etc...

So likely what a working implementation of this PR would look like for be a combination of splitting the ranges of commits, partially getting the data, and setting a timeout inbetween chunks to ensure that we are not hammering to API

Comment (codebytere) [reply branch-diff.js]: oh oh gotcha will do

**Comment by MylesBorins** [comment]
I just attempted to run

`branch-diff v8.x-staging upstream/v10.x --exclude-label semver-major,semver-minor,dont-land-on-v8.x,backport-requested-v8.x,backported-to-v8.x,baking-for-lts --filter-release` with this commit landed on my local branch diff and I still triggered the anti abuse trigger for github.

Would you be able to test this locally and figure out what magic number for a timeout is necessary to avoid the abuse mechanism

**Comment by joyeecheung** [comment]
I think the limit is triggered when you query the API without a personal access token? Orthogonal to this, would it help if we migrate branch-diff and changelog-maker to node-core-utils and use the token from ncurc?

**Comment by codebytere** [comment]
@joyeecheung sure, i'd be happy to take that on! I think centralizing a lot of these tools for collaborators increases their discoverability anyways.

**Pull request closed without merging**